products/compute/api.yaml

```diff
@@ -9324,6 +9324,176 @@ objects:
+        required: true
+        input: true
+      - !ruby/object:Api::Type::String
+        name: 'ipAddress'
+        description: |
+          IP address of the interface inside Google Cloud Platform.
+          Only IPv4 is supported.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'peerIpAddress'
+        description: |
+          IP address of the BGP interface outside Google Cloud Platform.
+          Only IPv4 is supported.
+        required: true
+        input: true
+      - !ruby/object:Api::Type::Integer
+        name: 'peerAsn'
+        description: |
+          Peer BGP Autonomous System Number (ASN).
+          Each BGP interface may use a different value.
+        required: true
+        input: true
+      - !ruby/object:Api::Type::Integer
+        name: 'advertisedRoutePriority'
+        description: |
+          The priority of routes advertised to this BGP peer.
+          Where there is more than one matching route of maximum
+          length, the routes with the lowest priority value win.
+      - !ruby/object:Api::Type::Enum
+        name: advertiseMode
+        description: |
+          User-specified flag to indicate which mode to use for advertisement.
+          Valid values of this enum field are: `DEFAULT`, `CUSTOM`
+        values:
+          - :DEFAULT
+          - :CUSTOM
+        default_value: :DEFAULT
+        input: true
+      - !ruby/object:Api::Type::Array
+        name: advertisedGroups
+        description: |
+          User-specified list of prefix groups to advertise in custom
+          mode, which can take one of the following options:
+
+          * `ALL_SUBNETS`: Advertises all available subnets, including peer VPC subnets.
+          * `ALL_VPC_SUBNETS`: Advertises the router's own VPC subnets.
+          * `ALL_PEER_VPC_SUBNETS`: Advertises peer subnets of the router's VPC network.
+
+
+          Note that this field can only be populated if advertiseMode is `CUSTOM`
+          and overrides the list defined for the router (in the "bgp" message).
+          These groups are advertised in addition to any specified prefixes.
+          Leave this field blank to advertise no custom groups.
+        send_empty_value: true
+        item_type: Api::Type::String
+        input: true
+      - !ruby/object:Api::Type::Array
+        name: advertisedIpRanges
+        description: |
+          User-specified list of individual IP ranges to advertise in
+          custom mode. This field can only be populated if advertiseMode
+          is `CUSTOM` and is advertised to all peers of the router. These IP
+          ranges will be advertised in addition to any specified groups.
+          Leave this field blank to advertise no custom IP ranges.
+        send_empty_value: true
+        item_type: !ruby/object:Api::Type::NestedObject
+            properties:
+              - !ruby/object:Api::Type::String
+                name: range
```

Comment:
This was changed to Required in 3.0.0 (sending an advertisedIpRange with no range doesn't really make sense semantically)

Reply:
Done. Also removed the `send_empty_value: true` as this doesn't make sense for a required field AFAIU.